### PR TITLE
Recursive mutating schema walker.

### PR DESCRIPTION
**Pull request by jiahuif**
This PR refactors `pkg/aggregator/mutating_walker.go` into a more generic schema walker.
plus fuzz testing for Schema mutations

### CODE REVIEW

File: pkg/schemamutation/walker.go [hunk 1]
```diff
@@ -48,14 +64,18 @@ func (w *mutatingReferenceWalker) walkSchema(schema *spec.Schema) *spec.Schema {
 		}
 	}
 
-	if r := w.walkRefCallback(&schema.Ref); r != &schema.Ref {
+	// Always callback on the whole schema first so that other callbacks
```
Comment (jpbetz) [inline pkg/schemamutation/walker.go:67]: The callbacks are pre-order DFS? That should probably be in the godoc of the walker of interface (and the call order is needs to be part of the interface contract).

File: pkg/schemamutation/walker.go [hunk 2]
```diff
@@ -14,28 +14,44 @@ See the License for the specific language governing permissions and
...
-	_ "net/http/pprof"
-
 	"k8s.io/kube-openapi/pkg/validation/spec"
 )
 
-// Run a walkRefCallback method on all references of an OpenAPI spec, replacing the values.
-type mutatingReferenceWalker struct {
-	// walkRefCallback will be called on each reference. Do not mutate the input, always create a copy first and return that.
-	walkRefCallback func(ref *spec.Ref) *spec.Ref
+// Walker runs callback methods on all references of an OpenAPI spec,
```
Comment (jpbetz) [inline pkg/schemamutation/walker.go:23]: very minor nit: s/methods/functions

**Review by jpbetz — COMMENTED** [review]
Minor doc comments, lgtm otherwise

File: pkg/schemamutation/walker.go (hunk 1 above)
Comment (jiahuif) [reply pkg/schemamutation/walker.go]: This is a DFS but callbacks are not recursive. The recursive part is with `WalkSchema` down below.

Added comments about the call ordering in the type and impl.

**Comment by apelisse** [comment]
/lgtm

Thanks!

**Pull request merged**